utils/poap-manager.ts

```diff
@@ -0,0 +1,115 @@
+import {DesmosClient, OfflineSignerAdapter, SigningMode} from "@desmoslabs/desmjs";
+import {program} from "commander";
+import * as Config from "./config"
+import {ExecuteMsg, InstantiateMsg, QueryMsg} from "@desmoslabs/contract-types/contracts/poap-manager";
+import {AccountData} from "@cosmjs/amino";
+import {parseCWTimestamp, parseIpfsUri} from "./cli-parsing-utils";
+
+async function main() {
+    const signer = await OfflineSignerAdapter.fromMnemonic(SigningMode.DIRECT, Config.mnemonic);
+    const client = await DesmosClient.connectWithSigner(Config.rpcEndpoint, signer, {
+        gasPrice: Config.gasPrice
+    });
+    const account: AccountData = (await signer.getCurrentAccount()) as AccountData;
+
+    program.name("Poap Manager contract utils")
+        .description("Utility script to interact with the poap manager contract");
+
+    program.command("init")
+        .description("Initialize a new instance of a poap manager contract")
+        .requiredOption("--code-id <code-id>", "Id of the contract to initialize", parseInt)
+        .requiredOption("--poap-code-id <poap-code-id>", "Id of the poap contract that will be initialized from the poap manager contract", parseInt)
+        .requiredOption("--cw721-code-id <cw721-code-id>", "Id of the cw721 contract that will be initialized from the poap contract", parseInt)
+        .requiredOption("--name <name>", "Poap name")
+        .requiredOption("--symbol <symbol>", "Poap symbol")
+        .requiredOption("--start <start>", "Datetime when the event will start in RFC3339 format (2022-12-31:10:00:00)", parseCWTimestamp)
+        .requiredOption("--end <end>", "Date when the event will end in RFC3339 format (2022-12-31:10:00:00)", parseCWTimestamp)
+        .requiredOption("--address-limit <address_limit>", "Max number of poap that an user can receive", parseInt)
+        .requiredOption("--poap-uri <poap_uri>", "IPFS uri where the event's metadata are stored", parseIpfsUri)
+        .option("--creator <creator>", "Bech32 address of who is creating the event", account!.address)
+        .option("--minter <minter>", "Bech32 address of who will have the minting rights", account!.address)
+        .option("--admin <admin>", "Bech32 address of who will have the contract admin rights", account!.address)
+        .action(async (options) => {
+            console.log(`Initializing contract with code ${options.codeId}`);
+            let instantiateMsg: InstantiateMsg = {
+                admin: options.admin,
+                poap_code_id: options.poapCodeId.toString(),
+                poap_instantiate_msg: {
+                    admin: options.admin,
+                    minter: options.minter,
```

Comment:
`minter` would be set as the poap manager contract address forcely. I think we can set it as "contract-address" hardcoded string, then remove minter option. What do you think?

Reply:
Yes totally forgot about that, I remove the `minter` option and put the hardcoded string.